doc/serial-timeout-troubleshoot.md

```diff
@@ -0,0 +1,80 @@
+# Troubleshooting "serial timeout" errors
+
+Several nRF Connect Desktop applications use serial ports to communicate with
+development kits and dongles. nRF Connect Desktop also uses serial ports to
+perform DFU (Device Firmware Update) when a devkit/dongle is running a 
+[DFU bootloader](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.sdk5.v15.0.0/sdk_app_serial_dfu_bootloader.html?cp=4_0_0_4_3_4).
+
+Serial ports might be [UARTs](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.nrf52840.ps/uart.html?cp=2_0_0_5_32)
+on the devkits connected to a RS-232 port on a computer,
+or they might be emulated through USB CDC ACM (also known as "Virtual COM port",
+"serial over USB" or other names). The later happens when using 
+[USB ACM CDC on the IMCU](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.nrf52/dita/nrf52/development/preview_dev_kit/vir_com_port.html?cp=2_1_5_4_1),
+or when using [USB functionality from the Nordic SDK](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.sdk5.v15.0.0/lib_usbd_class_cdc.html?cp=4_0_0_3_51_8_3)
+on SoCs that implement USB (e.g. the nRF52840).
+
+While serial ports are ubiquitous, serial port transport lacks any protocol
+metadata or capabilities for request-response higher-level protocols. Combined
+with the fact that modern desktop operating systems still allow for legacy
+serial ports, this can trigger timeouts.
+
+This document collects some known scenarios and workarounds for these timeout 
+errors.
+
+# Scenarios where these errors happen
+
+
+## DevKit/dongle not running bootloader
+
+When using a nRF52840 devkit/dongle and performing a DFU operation with 
+nRF Connect Desktop, a `Timeout while reading from serial transport.` error shows up.
+
+When using a nRF52840 devkit/dongle and performing a DFU operation with
+[nrfutil](https://github.com/NordicSemiconductor/pc-nrfutil/), a 
+`TypeError: 'NoneType' object is not iterable` error shows up.
+
+When performing a DFU operation, nrfutil *assumes* that the serial port
+specified is a devkit/dongle already in bootloader mode. nRF Connect Desktop (and
+the underlying [`nrf-device-setup-js`](https://github.com/NordicSemiconductor/nrf-device-setup-js) library)
+will try to perform a [DFU trigger](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.sdk5.v15.0.0/lib_dfu_trigger_usb.html) 
+first, but the DFU trigger might fail due to other reasons.
+
+In either case, the firmware in the nRF SoC is receiving DFU data through the
+serial port, but is not understanding it, and it's not replying to it.
+
+#### Workaround
+
+Ensure that the devkit/dongle is in bootloader mode.
+
+- For nRF52840 dongles:
```

Comment:
Is it necessary to unplug and re-plug while reset down? I've never done that, only pressed the reset while plugged in and it always worked. It is a bit more acrobatic to plug out-n-in, but if it covers some more cases or this is the official way... I'm just wondering.